model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -84,3 +90,36 @@ fun OrtResult.getDetectedLicensesWithCopyrights(
     omitExcluded: Boolean = true
 ): Map<String, Set<String>> =
     LicenseResolver(this, packageConfigurationProvider).getDetectedLicensesWithCopyrights(id, omitExcluded)
+
+/**
+ * Create [PackageConfiguration]s for the projects in this [OrtResult]. [PathExclude]s and [LicenseFindingCuration]s are
+ * taken from the [RepositoryConfiguration]. The [Provenance] is taken from the scan results. Projects without scan
+ * result and scan results without provenance information are ignored.
+ */
+fun OrtResult.getProjectConfigurations(): List<PackageConfiguration> {
+    fun VcsInfo.toMatcher() = VcsMatcher(
+        type = type,
+        url = url,
+        revision = resolvedRevision ?: revision,
+        path = path
+    )
+
+    return analyzer?.result?.projects.orEmpty().flatMap { project ->
+        getScanResultsForId(project.id).mapNotNull { scanResult ->
+            val sourceArtifactUrl = scanResult.provenance.sourceArtifact?.url
+            val vcs = scanResult.provenance.vcsInfo?.toMatcher()
+
+            if (sourceArtifactUrl == null && vcs == null) {
+                null
+            } else {
+                PackageConfiguration(
+                    id = project.id,
+                    sourceArtifactUrl = sourceArtifactUrl,
```

Comment:
The constructor throws in case both `sourceArtifactUrl` and `vcs` are `not null`.

Reply:
Yes, and that's fine, right? Not sure what you are up to.

Reply:
It should be ensure that this never happens.

Reply:
That cannot happen, because the provenance constructor ensures that only one of them is not null. And if it could happen I think it would be correct to run into the exception. However, with the change above this is now obsolete.